src/components/marker-chart/Canvas.js

```diff
@@ -129,6 +151,27 @@ class MarkerChartCanvas extends React.PureComponent<Props, State> {
       this.drawMarkers(ctx, hoveredItem, startRow, endRow);
       this.drawSeparatorsAndLabels(ctx, startRow, endRow);
     }
+
+    // For Highlighting the rightClickedRow
+    if (rightClickedRow !== undefined) {
```

Comment:
I realized that when right clicking we always go to the "full draw" branch, that is the `else` branch above. Therefore i think it would be a good idea to have this code there instead, something like this:
```js
} else {
  // 1. fillRect
  // 2. draw seperators and labels
  // 3. if (right clicked marker) highlightRow
  // 4. draw markers
}
```
Then we can remove this if block here.
See also the comment above.

Reply:
In that case, the highlighting goes away if as soon as we move the pointer away from the right-clicked marker. Hence it's in a different if block.